### PR TITLE
Orion Locking Indicator LED Support

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

Was going through my open branches and I forgot that Marcus/Keebology and I figured this out! PR-ing it!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/duck/orion/v3/matrix.c [hunk 1]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
-  PORTB |=  0b00010001;
-  DDRE  |=  0b01000000;
-  PORTE &= ~0b01000000;
+  
+}
+
+void indicator_init_ports(void) {
+
+  // Num LED
+  DDRB |=  (1<<4);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
  setPinOutput(B4); 
```

File: keyboards/duck/orion/v3/matrix.c [hunk 2]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
+}
+
+void indicator_init_ports(void) {
+
+  // Num LED
+  DDRB |=  (1<<4); 
+  PORTB &= ~(1<<4); 
+
+  // Caps Lock
+  DDRB |= (1<<0);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
  setPinOutput(B0); 
```

File: keyboards/duck/orion/v3/matrix.c [hunk 3]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
+  // Num LED
+  DDRB |=  (1<<4); 
+  PORTB &= ~(1<<4); 
+
+  // Caps Lock
+  DDRB |= (1<<0); 
+  PORTB &= ~(1<<0); 
+
+  // Scroll Lock
+  DDRD |= (1<<7);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
 setPinOutput(D7); 
```

File: keyboards/duck/orion/v3/matrix.c [hunk 4]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
-  DDRE  |=  0b01000000;
-  PORTE &= ~0b01000000;
+  
+}
+
+void indicator_init_ports(void) {
+
+  // Num LED
+  DDRB |=  (1<<4); 
+  PORTB &= ~(1<<4);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
```

File: keyboards/duck/orion/v3/matrix.c [hunk 5]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
+
+void indicator_init_ports(void) {
+
+  // Num LED
+  DDRB |=  (1<<4); 
+  PORTB &= ~(1<<4); 
+
+  // Caps Lock
+  DDRB |= (1<<0); 
+  PORTB &= ~(1<<0);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
```

File: keyboards/duck/orion/v3/matrix.c [hunk 6]
```diff
@@ -54,18 +54,26 @@ void matrix_scan_user(void) {
...
+  DDRB |=  (1<<4); 
+  PORTB &= ~(1<<4); 
+
+  // Caps Lock
+  DDRB |= (1<<0); 
+  PORTB &= ~(1<<0); 
+
+  // Scroll Lock
+  DDRD |= (1<<7); 
+  PORTD &= ~(1<<7);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]:
```suggestion
```

File: keyboards/duck/orion/v3/matrix.c [hunk 7]
```diff
@@ -54,18 +54,23 @@ void matrix_scan_user(void) {
...
+void indicator_init_ports(void) {
+
+  // Num LED
+  setPinOutput(B4); 
+
+  // Caps Lock
+  setPinOutput(B0); 
+
+  // Scroll Lock
+ setPinOutput(D7);
```
Comment (drashna) [inline keyboards/duck/orion/v3/matrix.c]: This indent is killing me

File: keyboards/duck/orion/v3/v3.c [hunk 8]
```diff
@@ -16,48 +16,26 @@
...
+// only able to control them ALL with the current default mechanisms. 
 
-  indicator_leds_set(status);
+// Locking indicator LEDs
+// The Duck Orion V3 has 3 locking indicator LEDs and are located to the right
+// of the Escape key. 
+bool led_update_kb(led_t led_state) {
+    if(led_update_user(led_state)) {
+        writePin(B0, !led_state.caps_lock);
+        writePin(B4, !led_state.scroll_lock);
```
Comment (noroadsleft) [inline keyboards/duck/orion/v3/v3.c]: Shouldn't this be:

```suggestion
        writePin(B4, !led_state.num_lock);
```

File: keyboards/duck/orion/v3/keymaps/default/keymap.c [hunk 9]
```diff
@@ -17,10 +17,18 @@
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 [0] = LAYOUT_tkl_ansi(\
```
Comment (fauxpark) [inline keyboards/duck/orion/v3/keymaps/default/keymap.c]:
```suggestion
[0] = LAYOUT_tkl_ansi(
```

File: keyboards/duck/orion/v3/keymaps/default/keymap.c [hunk 10]
```diff
@@ -17,10 +17,18 @@
...
-  KC_LSFT,    KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,        KC_RSFT,            KC_UP,        \
-  KC_LCTL,KC_LGUI,KC_LALT,                 KC_SPC,                                KC_RALT,KC_RGUI, KC_APP,KC_RCTL,  KC_LEFT,KC_DOWN,KC_RGHT) \
+  KC_ESC,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,           KC_PSCR,KC_SLCK,KC_PAUS, 
+  KC_GRV,   KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_MINS, KC_EQL,KC_BSPC,   KC_INS ,KC_HOME,KC_PGUP, 
+  KC_TAB,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_LBRC,KC_RBRC,KC_BSLS,   KC_DEL ,KC_END ,KC_PGDN, 
+  KC_CAPS,   KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,         KC_ENT,                          
+  KC_LSFT,    KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,        KC_RSFT,            KC_UP,        
+  KC_LCTL,KC_LGUI,KC_LALT,                 KC_SPC,                                KC_RALT,KC_RGUI, MO(1),KC_RCTL,  KC_LEFT,KC_DOWN,KC_RGHT),
+
+[1] = LAYOUT_tkl_ansi(\
```
Comment (fauxpark) [inline keyboards/duck/orion/v3/keymaps/default/keymap.c]:
```suggestion
[1] = LAYOUT_tkl_ansi(
```

File: keyboards/duck/orion/v3/readme.md [hunk 11]
```diff
@@ -4,7 +4,7 @@ Non official firmware for custom TKL Korean keyboard made by Duck.
 Group buy was run December 2018 via [geekhack](https://geekhack.org/index.php?topic=98581.0) with 100 keyboards total. 
 
 Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)
```
Comment (fauxpark) [inline keyboards/duck/orion/v3/readme.md]:
```suggestion
* Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)
```

File: keyboards/duck/orion/v3/readme.md [hunk 12]
```diff
@@ -4,7 +4,7 @@ Non official firmware for custom TKL Korean keyboard made by Duck.
 Group buy was run December 2018 via [geekhack](https://geekhack.org/index.php?topic=98581.0) with 100 keyboards total. 
 
 Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)  
-Hardware Supported: Duck Eagle/Viper PCB Ver 2.0, Atmega32u4  
+Hardware Supported: Duck Orion V3 PCB Ver 2.1
```
Comment (fauxpark) [inline keyboards/duck/orion/v3/readme.md]:
```suggestion
* Hardware Supported: Duck Orion V3 PCB Ver 2.1
```

File: keyboards/duck/orion/v3/readme.md [hunk 13]
```diff
@@ -4,7 +4,7 @@ Non official firmware for custom TKL Korean keyboard made by Duck.
 Group buy was run December 2018 via [geekhack](https://geekhack.org/index.php?topic=98581.0) with 100 keyboards total. 
 
 Keyboard Maintainer: [MechMerlin](https://github.com/mechmerlin)  
-Hardware Supported: Duck Eagle/Viper PCB Ver 2.0, Atmega32u4  
+Hardware Supported: Duck Orion V3 PCB Ver 2.1  
 Hardware Availability: Wait until GB of the next revision
```
Comment (fauxpark) [inline keyboards/duck/orion/v3/readme.md]:
```suggestion
* Hardware Availability: Wait until GB of the next revision
```

**Pull request merged**